src/Protocol/Diaspora.php

```diff
@@ -772,55 +784,51 @@ private static function validPosting(array $msg)
 	/**
 	 * Fetches the public key for a given handle
 	 *
-	 * @param string $handle The handle
+	 * @param WebFingerUri $uri The handle
 	 *
 	 * @return string The public key
-	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws InternalServerErrorException
 	 * @throws \ImagickException
 	 */
-	private static function key(string $handle = null): string
+	private static function key(WebFingerUri $uri): string
 	{
-		$handle = strval($handle);
-
-		Logger::notice('Fetching diaspora key', ['handle' => $handle, 'callstack' => System::callstack(20)]);
-
-		$fcontact = FContact::getByURL($handle);
-		if (!empty($fcontact['pubkey'])) {
-			return $fcontact['pubkey'];
+		Logger::notice('Fetching diaspora key', ['handle' => $uri->getAddr(), 'callstack' => System::callstack(20)]);
+		try {
+			return DI::dsprContact()->getByAddr($uri)->pubKey;
+		} catch (HTTPException\NotFoundException|\InvalidArgumentException $e) {
+			return '';
```

Comment:
No logging ?

Reply:
There was none before, not sure it would be helpful.